artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerConsumerImpl.java

```diff
@@ -1068,6 +1074,31 @@ private void deliverStandardMessage(final MessageReference ref, final Message me
       }
    }
 
+   private void applyPrefixForLegacyConsumer(Message message) {
+      /**
+       * check to see if:
+       * 1) This is a "core" connection
+       * 2) The "core" connection belongs to a JMS client
+       * 3) The JMS client is an "old" client which needs address prefixes
+       *
+       * If 1, 2, & 3 are true then apply the "old" prefix for queues and topics as appropriate.
+       */
+      if (session.getRemotingConnection() instanceof CoreRemotingConnection) {
```

Comment:
Can you please cache this during construction, and only apply this if the cached boolean is set?

This is part of the hot path for delivering messages.. so any gain counts.